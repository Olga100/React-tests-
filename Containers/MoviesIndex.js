import React from 'react';
import {connect} from 'react-redux';
import MovieList from './MovieList';
import MovieDetails from './MovieDetails';
import MainHeader from '../Components/MainHeader';
import {Link} from 'react-router';
import {setAdmin, fetchMovies} from '../Actions/Actions';


export class MoviesIndexView extends React.Component {

    componentDidMount() {
       this.props.loadMovies();       //СРАЗУ ПРИ старте загрузить муви
    }

    renderModeButton() {
        if (!this.props.isAdmin) {
            return <button type="button" onClick={this.props.setAdmin} className="admin">Admin</button>
        }
        else {
            return <button type="button" onClick={this.props.setUser} className="user">User</button>
        }
    }

    renderModeCreate() {

        if (this.props.isAdmin) {

            return (
                <Link to="/createMovie" activeClassName="active">
                    <button type="button" className="create">Create Movie</button>
                </Link>
            )
        }
    }

    render() {
        return (
            <div className="App">

                <MainHeader/>

                <div className="header">

                    {this.renderModeCreate()}

                    <h2>Movies</h2>
                    <div className="buttonMode">
                        {this.renderModeButton()}
                    </div>
                </div>

                <div className="movie-list-container">
                    <MovieList />
                </div>

                <div className="movie-details-container">
                    <MovieDetails />
                </div>

                <div className="clear"></div>

                <div className="footer"><h2> EPAM</h2></div>
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => ({
    isAdmin: state.viewState.isAdmin //всё что до двоеточия(мувис, каррентмувис сетАлмин - падают в пропсы, т.е. обращаемся мы к ним через this.props.
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    setAdmin: () => dispatch(setAdmin(true)),
    setUser: () => dispatch(setAdmin(false)),
    loadMovies: () => dispatch(fetchMovies())
});

const MoviesIndex = connect(mapStateToProps, mapDispatchToProps)(MoviesIndexView);

export default MoviesIndex;

