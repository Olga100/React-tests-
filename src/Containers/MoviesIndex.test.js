import React from 'react';
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {MoviesIndexView} from './MoviesIndex';


describe('<MoviesIndexView/>', () => {

    it('Renders  main components for movies page', () => {

        const loadMovies = jest.fn();

        const mockStore = configureMockStore();
        const store = mockStore({
            viewState: {
                isAdmin: true,
                filter: "",
                sortBy: "rating"
            },
            movies: []
        });

        const component = renderer.create(
            <Provider store={store}>
                <MoviesIndexView isAdmin="true" loadMovies={loadMovies}/>
            </Provider>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });


});

