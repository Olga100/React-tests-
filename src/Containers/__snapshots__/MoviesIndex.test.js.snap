// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesIndexView/> Renders  main components for movies page 1`] = `
<div
  className="App"
>
  <div
    className="mainHeader"
  >
    <span
      className="nav"
    >
      <a
        className="navLink"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </span>
    <span
      className="nav"
    >
      <a
        className="navLink"
        onClick={[Function]}
        style={Object {}}
      >
        Movies
      </a>
    </span>
    <span
      className="nav"
    >
      <a
        className="navLink"
        onClick={[Function]}
        style={Object {}}
      >
        About
      </a>
    </span>
  </div>
  <div
    className="header"
  >
    <a
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className="create"
        type="button"
      >
        Create Movie
      </button>
    </a>
    <h2>
      Movies
    </h2>
    <div
      className="buttonMode"
    >
      <button
        className="user"
        onClick={undefined}
        type="button"
      >
        User
      </button>
    </div>
  </div>
  <div
    className="movie-list-container"
  >
    <div
      className="leftPart"
    >
      <div
        className="FieldSortSearch"
      >
        <p>
          Sort movies
        </p>
        <div>
          <button
            className="sortByLikies"
            onClick={[Function]}
            type="button"
          >
            <small>
               by likies
            </small>
          </button>
          <button
            className="sortByRating"
            onClick={[Function]}
            type="button"
          >
            <small>
               by rating
            </small>
          </button>
        </div>
        <div
          className="fieldInput"
        >
          <button
            className="submit"
            onClick={[Function]}
            type="submit"
          >
            ðŸ”Ž
          </button>
          <input
            className="input"
            onChange={[Function]}
            placeholder="Search by name"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-details-container"
  />
  <div
    className="clear"
  />
  <div
    className="footer"
  >
    <h2>
       EPAM
    </h2>
  </div>
</div>
`;
